Yiheng (#1904)
kamranahmedse/developer-roadmap: content/roadmaps/110-java/content/107-testing-java-apps/100-mocking.md

> # Mocking
> Mocking is a way of removing external dependencies from a unit test so as to create a sense of an entire controlled environment.The traditional way of using mocks invovles mocking all other classes that interact with the class that we want to test. The common targets for mocking are
> - Database connections
> - Web services
> - Classes that are slow
> - Classes with side effects
> - Classes with non-deterministic behavior
> 
> <ReferenceGroupTitle>Reference Resource</ResourceGroupTitle>
> <BadgeLink colorScheme='blue' badgeTest='Official Website' href='https://site.mockito.org/'>Mocking Website</BadgeLink>

Reviewer: ```suggestion
# Mocking

Mocking removes external dependencies from a unit test to create a sense of an entire controlled environment. The traditional method of mocks involves mocking all other classes that interact with the class we want to test. The common targets for mocking are:

- Database connections
- Web services
- Slow Classes
- Classes with side effects
- Classes with non-deterministic behavior

<ReferenceGroupTitle>Reference Resource</ResourceGroupTitle>
<BadgeLink colorScheme='purple' badgeTest='Visit' href='https://site.mockito.org/'>Mockito - Mocking Framework for Java</BadgeLink>
```
